uqcsbot/scripts/wolfram.py

```diff
@@ -0,0 +1,192 @@
+        return
+
+    parent_message = thread_parent['messages'][0]  # Limit=1 was used so the first (and only) message is what we want
+    # If the threads parent wasn't by a bot ignore it
+    if parent_message.get('subtype') != 'bot_message':
+        return
+
+    # Finally, we have to check that this is a Wolfram replyable message
+    # It is rare we would reach this point and not pass as who replies to a bot in a thread for another reason?
+    parent_attachments = parent_message['attachments'][0]  # The message is formatted with only one attachment
+    parent_fallback = parent_attachments['fallback']
+    if 'WolframCanReply' not in parent_fallback:
+        return
+
+    # Now we can grab the conversation_id from the message and get the new question (s only sometimes appears)
+    # Recall the format of the fallback "identifier hostname s_output conversationID"
+    _, reply_host, s_output, conversation_id = parent_fallback.split(' ')
+    new_question = evt['text']  # This is the value of the message that triggered this whole response
+    s_output = '' if s_output == 'None' else s_output
+
+    # Slack annoyingly formats the reply_host link so we have to extract what we want:
+    reply_host = reply_host[1:-1].split('|')[0]
+
+    # Now we can ask Wolfram for the next answer:
+    api_url = f'{reply_host}/api/v1/conversation.jsp?'
+    params = {'appid': APP_ID, 'i': new_question, 'conversationid': conversation_id, 's': s_output}
+    http_response = await bot.run_async(requests.get, api_url, params=params)
+
+    if http_response.status_code != requests.codes.ok:
+        bot.post_message(channel, "There was a problem getting the response", thread_ts=thread_ts)
+        return
+
+    # Convert to json and check for an error
+    wolfram_answer = json.loads(http_response.content)
+    if 'error' in wolfram_answer:
+        bot.post_message(channel, wolfram_answer['error'], thread_ts=thread_ts)
+        return
+
+    # Otherwise grab the new stuff and post the reply.
+    reply = wolfram_answer['result']
+    conversation_id = wolfram_answer['conversationID']
+    reply_host = wolfram_answer['host']
+    s_output = wolfram_answer.get('s', None)
+
+    bot.post_message(channel, reply, thread_ts=thread_ts)
+
+    # Update the old fallback to reflect the new state of the conversation
+    parent_attachments['fallback'] = f'WolframCanReply {reply_host} {s_output} {conversation_id}'
+
+    bot.api.chat.update(channel=channel, attachments=[parent_attachments], ts=thread_ts)
+
+
+async def short_answer(search_query: str):
+    """
+    This uses wolfram's short answers api to just return a simple short plaintext response.
+
+    This is used if the conversation api fails to get a result (for instance !wolfram pineapple is not a great
+    conversation starter but may be interesting.
+    """
+    api_url = r"http://api.wolframalpha.com/v1/result?"
```

Comment:
I notice you use v1 here and v2 somewhere else, any reason?

Reply:
and why `r""`?

Reply:
I'd upgrade to latest API if it doesn't change anything, just for consistency and longevity :^)